### PR TITLE
Fix RMSE bug in metrics.py

### PR DESCRIPTION
**Pull request by am2145**
## Description
This PR edits the `RMSEMetric` in `metrics.py` to address the bug identifed in issue #577, namely that the RMSE and MAE metrics were observed to return the same value in the example training notebook. The reason for this bug seems to be due to the inheritance of the `RMSEMetric` class. In the current version, the .`sqrt()` operation appears to occur before the mean reduction in metrics, resulting in it effectly computing MAE. To circumvent this, @hwpang and I modified the class to have its own `__call__() `method. The loss functions in` loss.py `  do work as intended and so are untouched. 

## Questions
The implementation works but there may be a more elegant solution that better follows the intended class inheritance structure. 

## Relevant issues
#577

### CODE REVIEW

File: chemprop/nn/metrics.py [hunk 1]
```diff
@@ -74,8 +74,20 @@ class MSEMetric(MSELoss, Metric):
...
+        w_s: Tensor,
+        w_t: Tensor,
+        lt_mask: Tensor,
+        gt_mask: Tensor,
+    ):
+        return self.forward(preds, targets, mask, lt_mask, gt_mask)[mask].mean().sqrt()
+
     def forward(self, *args, **kwargs) -> Tensor:
-        return super().forward(*args, **kwargs).sqrt()
+        return super().forward(*args, **kwargs)
```
Comment (davidegraff) [inline chemprop/nn/metrics.py:90]: you can delete this function entirely if it only calls the `super()`

**Comment by davidegraff** [comment]
You've got the solution correct. The previous implementation calculated the quantity:

```math
\frac{1}{N * T} \sum^N_i \sum^T_j \sqrt{\ell_{ij}}
```

which is not equivalent to the RMSE:

```math
\sqrt{\frac{1}{N * T} \sum^N_i \sum^T_j \ell_{ij}}
```

File: chemprop/nn/metrics.py (hunk 1 above)
Comment (am2145) [reply chemprop/nn/metrics.py]: Got it. I have removed the forward function() and implemented the call to `super()` in the '__call__()' method itself.

**Review by davidegraff — APPROVED** [review]
👍

**Comment by am2145** [comment]
Thank you for the feedback and checking the implementation @davidegraff. I changed the class as suggested above and have opened the PR as I believe this should be ready to go. Thank you to @hwpang as well for helping resolve this.

**Pull request merged**